dbms/src/Common/ConcurrentIOQueue.h

```diff
@@ -48,9 +48,17 @@ class ConcurrentIOQueue
 
     /// blocking function.
     /// Just like MPMCQueue::push.
-    MPMCQueueResult push(T && data)
+    template <typename U>
+    MPMCQueueResult push(U && data)
     {
-        return mpmc_queue.push(std::move(data));
+        return mpmc_queue.push(std::forward<U>(data));
+    }
+
+    /// Just like MPMCQueue::tryPush.
+    template <typename U>
+    MPMCQueueResult tryPush(U && data)
+    {
+        return mpmc_queue.tryPush(std::forward<U>(data));
```

Comment:
why change here?

Reply:
> why change here?

I will revoke the change of `push`.

For `tyPush`, not always we wan't to pass value by std::move, as tryPush may fail when queue is full. So, sometimes we want to pass value by reference when using tryPush